mgr/src/server/plugins/volume_rename.cr

```diff
@@ -19,19 +19,54 @@ post "/api/v1/pools/:pool_name/volumes/:volume_name/rename" do |env|
     halt(env, status_code: 400, response: ({"error": "Volume does not exist."}.to_json))
   end
 
-  if new_pool_name != pool_name
-    halt(env, status_code: 400, response: ({"error": "Volume rename outside the pool #{pool_name} is not supported."}.to_json))
+  new_pool = Datastore.get_pool(new_pool_name)
+
+  if new_pool.nil?
+    halt(env, status_code: 400, response: ({"error": "Target Pool #{new_pool_name} does not exist."}.to_json))
   end
 
-  if volume_name == new_volname
-    halt(env, status_code: 400, response: ({"error": "Existing & New volname are the same."}.to_json))
+  if Datastore.volume_name_exists_by_pool_id?(new_volname, new_pool.id)
+    halt(env, status_code: 400, response: ({"error": "Volume #{new_volname} already exists in target Pool #{new_pool_name}"}.to_json))
   end
 
   if volume.state == "Started"
     halt(env, status_code: 400, response: ({"error": "Volume should be stopped before renaming."}.to_json))
   end
 
-  Datastore.rename_volume(pool.id, volume.id, new_volname)
+  transfer_volume = pool_name != new_pool_name ? true : false
+  if transfer_volume
+    if Datastore.node_part_of_other_volume?(pool.id, volume.id)
+      halt(env, status_code: 400, response: ({"error": "Node(s) are part of other volume(s) in the current pool."}.to_json))
+    end
+
+    # Get List of nodes which are part of this volume, which needs to be updated to new pool.
+    node_id_list = Datastore.list_of_nodes_part_of_curr_volume(pool.id, volume.id)
+
+    # Update pool_name in .info file at every local_data_nodes
+    puts "here above participating nodes"
+    nodes = participating_nodes(pool_name, volume)
+    puts "nodes: #{nodes}"
+    resp = dispatch_action(
+      ACTION_NODE_POOL_RENAME,
+      pool_name,
+      nodes,
+      {"new_pool_name": new_pool_name, "old_pool_name": pool_name}.to_json,
+    )
+
+    if !resp.ok
+      halt(env, status_code: 400, response: ({"error": "Failed to rename volume"}.to_json))
+    end
+
```

Comment:
Better to create a single function from datastore that does all the below statements as Transaction. Refer Volume create query execution. Good thing about transaction is it rollsback all the the changes if any one of the query in transaction fails.

Reply:
Done. Should node update be included in this single method in volumes.cr or is it good if it remains in nodes.cr

Reply:
Yes, all Db operations can be combined together. Transactions will give assurance of all or none if it fails in between.

Reply:
For now this is fine, you can add a TODO entry in code to pick the task later

Reply:
Added a Good First Task TODO